pkg/validation/validate_name.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var firstCharRegexp = regexp.MustCompile(`^[a-z0-9]`)
+var nameRegexp = regexp.MustCompile(`^[a-z0-9_.-]+$`)
+var lastCharRegexp = regexp.MustCompile(`[a-z0-9]$`)
+
+// ValidateName validates names to ensure they follow the Kubernetes
+// conventions for how names are constructed. For more about names,
+// see: k8s.io/docs/concepts/overview/working-with-objects/names/
+func ValidateName(n string) error {
```

Comment:
I didn't compare them but you may want to look at `k8s.io/kubernetes/pkg/apis/core/validation/validation.go:ValidateObjectMeta` and see if this function covers the same properties as name validation in there.